qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -132,44 +160,81 @@ def evolve(self, evolution_problem: TimeEvolutionProblem) -> TimeEvolutionResult
-        if not isinstance(hamiltonian, (Pauli, PauliSumOp)):
+        if not isinstance(hamiltonian, (Pauli, PauliSumOp, SparsePauliOp)):
             raise ValueError(
                 f"TrotterQRTE only accepts Pauli | PauliSumOp, {type(hamiltonian)} provided."
             )
-        # the evolution gate
-        evolution_gate = PauliEvolutionGate(
-            hamiltonian, evolution_problem.time, synthesis=self.product_formula
-        )
+        t_param = evolution_problem.t_param
+        if t_param is not None and not _get_parameters(hamiltonian.coeffs)==ParameterView([t_param]):
+            raise ValueError(
+                "Hamiltonian time parameter does not match ``evolution_problem.t_param`` "
+                "or contains multiple parameters"
+            )
+        
+
+        # make sure PauliEvolutionGate does not implement more than one Trotter step
+        dt = evolution_problem.time / self.num_timesteps
 
         if evolution_problem.initial_state is not None:
             initial_state = evolution_problem.initial_state
-            evolved_state = QuantumCircuit(initial_state.num_qubits)
-            evolved_state.append(initial_state, evolved_state.qubits)
-            evolved_state.append(evolution_gate, evolved_state.qubits)
-
         else:
             raise ValueError("``initial_state`` must be provided in the ``TimeEvolutionProblem``.")
 
-        evaluated_aux_ops = None
+        evolved_state = QuantumCircuit(initial_state.num_qubits)
+        evolved_state.append(initial_state, evolved_state.qubits)
+        
         if evolution_problem.aux_operators is not None:
-            evaluated_aux_ops = estimate_observables(
-                self.estimator,
-                evolved_state,
-                evolution_problem.aux_operators,
-                None,
-                evolution_problem.truncation_threshold,
+            observables = []
+            observables.append(
+                estimate_observables(
+                    self.estimator,
+                    evolved_state,
+                    evolution_problem.aux_operators,
+                    None,
+                    evolution_problem.truncation_threshold,
+                )
+            )
+        else:
+            observables = None
+
+        if t_param is None:
+            # the evolution gate
+            single_step_evolution_gate = PauliEvolutionGate(
+                hamiltonian, dt, synthesis=self.product_formula
             )
 
-        return TimeEvolutionResult(evolved_state, evaluated_aux_ops)
+        for n in range(self.num_timesteps):
+            # if hamiltonian is time-dependent, bind new time-value at every step to construct
+            # evolution for next step
+            if t_param is not None:
+                time_value = (n + 1)*dt
+                parametrized_coeffs = copy.deepcopy(hamiltonian.coeffs)
+                bound_coeffs = _assign_parameters(parametrized_coeffs, [time_value])
```

Comment:
It seems a bit weird we have to copy and then `_assign_parameters` also returns a new value. Maybe a cleaner solution would be to add a `copy` argument to `assign_parameters` (``True`` per default) which does the copying?